File: src/main/java/com/mycompany/bookshop/model/Checkout.java
/*
 * |-------------------------------------------------
 * | Copyright © 2017 Colin But. All rights reserved.
 * |-------------------------------------------------
 */
package com.mycompany.bookshop.model;

import lombok.extern.slf4j.Slf4j;

import java.util.Arrays;

@Slf4j
public class Checkout {

    private BookShop bookShop;

    public Checkout(BookShop bookShop) {
        this.bookShop = bookShop;
    }

    public void checkoutBooks(Book... books) {
        Money totalBookCost = bookShop.buyBooks(Arrays.asList(books));
        log.info("Total Cost of books: {} is {}", books, totalBookCost);
    }
}

File: kotlin/src/main/kotlin/bookshop/Checkout.kt
/*
 * |-------------------------------------------------
 * | Copyright © 2019 Colin But. All rights reserved.
 * |-------------------------------------------------
 */
package bookshop

class Checkout(private val bookshop : BookShop) {

    internal fun checkoutBooks(vararg books : Book) {
        val totalBookCost = bookshop.buyBooks(books.toList())
        println("Total Cost of books: $books is $totalBookCost")
    }
}